### ISSUE TITLE
Output mismatch of duplicate torch.sigmoid nodes with and without optimization

### ISSUE THREAD

**Issue by Azyka** [microsoft/onnxruntime#18214]
### Describe the issue

ONNX opset version: 14
When 2 duplicate nodes(which have the same inputs and outputs) of torch.sigmoid are defined, the model produce wrong results with and without ort optimization.

### To reproduce

Model code:
```
import onnxruntime as ort
import onnx
import numpy as np
from numpy import testing
import torch

class Model0(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, *args):
        _args = args
        getitem = _args[0]
        getitem_1 = _args[1]
        add = torch.add(getitem_1, getitem)
        sigmoid = torch.sigmoid(add)
        sigmoid_1 = torch.sigmoid(add)
        to = sigmoid_1.to(dtype = torch.int64)
        clip = torch.clip(to, -1, 1)
        return (sigmoid, clip)

model_0 = Model0()
output_names_0 = ['v0_0', 'v6_0']
input_data_0 = np.array(4.742, dtype=np.float16)
input_data_1 = np.array(6.883, dtype=np.float16)
input_dict_0 = {'v4_0':input_data_0, 'v3_0':input_data_1}
inputs_0 = tuple(torch.from_numpy(v).to('cpu') for _, v in input_dict_0.items())
torch.onnx.export(model_0, inputs_0, '0.onnx', verbose=False, input_names=['v4_0', 'v3_0'], output_names=output_names_0, opset_version=14, do_constant_folding=False)

class Model1(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, *args):
        _args = args
        getitem = _args[0]
        getitem_1 = _args[1];  _args = None
        add = torch.add(getitem, getitem_1);  getitem = getitem_1 = None
        sigmoid = torch.sigmoid(add);  add = None
        to = sigmoid.to(dtype = torch.int64);  sigmoid = None
        clip = torch.clip(to, -1, 1);  to = None
        return (clip,)

model_1 = Model1()
output_names_1 = ['v6_0']
input_dict_1 = {'v0_0':input_data_0, 'v1_0':input_data_1}
inputs_1 = tuple(torch.from_numpy(v).to('cpu') for _, v in input_dict_1.items())
torch.onnx.export(model_1, inputs_1, '1.onnx', verbose=False, input_names=['v0_0', 'v1_0'], output_names=output_names_1, opset_version=14, do_constant_folding=False)

sess_options_0 = ort.SessionOptions()
sess_options_0.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_ALL
sess_0 = ort.InferenceSession('0.onnx',providers=['CPUExecutionProvider'],sess_options=sess_options_0)
sess_res_0 = sess_0.run(output_names_0, input_dict_0)
output_0 = dict(zip(output_names_0, sess_res_0))

sess_options_1 = ort.SessionOptions()
sess_options_1.graph_optimization_level = ort.GraphOptimizationLevel.ORT_ENABLE_ALL
sess_1 = ort.InferenceSession('1.onnx',providers=['CPUExecutionProvider'],sess_options=sess_options_1)
sess_res_1 = sess_1.run(output_names_1, input_dict_1)
output_1 = dict(zip(output_names_1, sess_res_1))
output_name_dict = {'v6_0': 'v6_0'}

print('=========================')
try:
    for tensor_name_0, tensor_name_1 in output_name_dict.items():
        testing.assert_allclose(output_0[tensor_name_0], output_1[tensor_name_1])
    print("onnxruntime_enable_opt does not trigger assertion")
except AssertionError as e:
    print("onnxruntime_enable_opt triggers assertion")
    print(e)
print('=========================')

sess_options_0 = ort.SessionOptions()
sess_options_0.graph_optimization_level = ort.GraphOptimizationLevel.ORT_DISABLE_ALL
sess_0 = ort.InferenceSession('0.onnx',providers=['CPUExecutionProvider'],sess_options=sess_options_0)
sess_res_0 = sess_0.run(output_names_0, input_dict_0)
output_0 = dict(zip(output_names_0, sess_res_0))

sess_options_1 = ort.SessionOptions()
sess_options_1.graph_optimization_level = ort.GraphOptimizationLevel.ORT_DISABLE_ALL
sess_1 = ort.InferenceSession('1.onnx',providers=['CPUExecutionProvider'],sess_options=sess_options_1)
sess_res_1 = sess_1.run(output_names_1, input_dict_1)
output_1 = dict(zip(output_names_1, sess_res_1))

print('=========================')
try:
    for tensor_name_0, tensor_name_1 in output_name_dict.items():
        testing.assert_allclose(output_0[tensor_name_0], output_1[tensor_name_1])
    print("onnxruntime_disable_opt does not trigger assertion")
except AssertionError as e:
    print("onnxruntime_disable_opt triggers assertion")
    print(e)
print('=========================')
```
Output:
```
=========================
onnxruntime_enable_opt triggers assertion

Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 1
Max relative difference: inf
 x: array(1, dtype=int64)
 y: array(0, dtype=int64)
=========================
=========================
onnxruntime_disable_opt triggers assertion

Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 1
Max relative difference: inf
 x: array(1, dtype=int64)
 y: array(0, dtype=int64)
=========================
```

### Urgency

This is an incorrect functionality implementation. It may cause severe bugs for those systems on the top of ORT.

### Platform

Linux

### OS Version

Ubuntu 22.04.3 LTS (x86_64)

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.15.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by cbourjau** [comment]
Can you please provide the model files (or better yet a single model) that reproduces the issue?

**Comment by Azyka** [comment]
Here is the 2 related model files. And I am sorry that I am not sure which model contains the bug. Despite model0 seems more suspicious, it somehow produces the correct answer, and model1 is the wrong one.
[bug_models.zip](https://github.com/microsoft/onnxruntime/files/13269696/bug_models.zip)
